src/PartEditor/PartEditor.ts

```diff
@@ -123,6 +130,58 @@ export class PartEditorProvider implements vscode.CustomTextEditorProvider {
     }
   }
 
+  private handleRequestOpNames() {
+    const K_DATA: string = 'data';
+    const K_EXIT: string = 'exit';
+    const K_ERROR: string = 'error';
+    // TODO integrate with Toolchain
+    const tool = '/usr/share/one/bin/circle-operator';
+    const toolargs = ['--name', this._modelFilePath];
+    let result: string = '';
+    let error: string = '';
+
+    let runPromise = new Promise<string>((resolve, reject) => {
+      let cmd = cp.spawn(tool, toolargs, {cwd: this._modelFolderPath});
+
+      cmd.stdout.on(K_DATA, (data: any) => {
+        let str = data.toString();
+        if (str.length > 0) {
+          result = result + str;
+        }
+      });
+
+      cmd.stderr.on(K_DATA, (data: any) => {
+        error = result + data.toString();
+        Logger.error('Partition', error);
+      });
+
+      cmd.on(K_EXIT, (code: any) => {
+        let codestr = code.toString();
+        if (codestr === '0') {
+          resolve(result);
+        } else {
+          let msg = 'Failed to load model: ' + this._modelFileName;
+          Balloon.error(msg);
+          reject(msg);
+        }
+      });
+
+      cmd.on(K_ERROR, (err) => {
+        Balloon.error('Failed to run circle-operator: ' + this._modelFileName);
```

Comment:
(Q) Is it okay not to call `reject()` here?

Reply:
Thanks for this, I'll check.

Reply:
added reject()